### PR TITLE
Reader for MRtrix images

### PR DESCRIPTION
**Pull request by ylep**
A reader for MRtrix images ([here is the documentation for the format](https://mrtrix.readthedocs.io/en/dev/getting_started/image_data.html)).

This is still a work in progress, this PR is mostly a note for myself at this stage.

Reading basic files works fine, but several things are known to be broken:

- [x] transformations
- [x] some (most) mif files of an actual diffusion MR study are not recognized

Also, some features are missing:

- [x] reading of gzip-compressed files (`.mif.gz`)
- [x] reading of `bit` datatype
- [x] reading of non-essential metadata fields, e.g. diffusion b-values and b-vectors, command history, MRtrix version... (not essential...)
- [x] support for split mif files (`.mih` + `.dat`, never seen a single one in the wild...)

Finally, I'd like an expert to review a few points:
- [x] the “capabilities” (in the `DataSourceInfo` sense)
- [x] the `storage_to_memory` matrix (should it be 3-dimensional or more in case of higher-dimension images?)

### CODE REVIEW

**Comment by ylep** [comment]
Hi @denisri , if you have time I'd love if you colud review this PR!

You will need to switch `aims-free` to the `mif_io` branch as well.

I can provide test images if you want, otherwise you can create them yourself easily, as MRtrix is available as a packages on Ubuntu 22.04:

```
sudo apt install mrtrix3
mrconvert test.nii test.mif
```

**Comment by denisri** [comment]
OK, I'm looking at it (still compiling)...

File: src/somamif/plugin/dummy.cc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// only needed on Mac bundle
+#include <soma-io/plugin/mifplugin.h>
+
+using namespace soma;
+
+namespace
+{
+  // force using libaimsdicom.dylib on Mac
```
Comment (denisri) [inline src/somamif/plugin/dummy.cc]: this comment is outdated ;)

File: src/somamif/plugin/mifplugin.cc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      return true;
+    }
+    bool mifinit __attribute__((unused)) = initmif();
+  }
+}
+
+MifPlugin::MifPlugin() : Plugin()
+{
+    vector<string>  exts(5);
+    exts[0] = "nii";
```
Comment (denisri) [inline src/somamif/plugin/mifplugin.cc]: exts should contain "mif" and not "nii" and others ?

**Review by denisri — APPROVED** [review]
Difficult to review every line of code... At least I have tried it, it works (for reading).
I have pushed a fix for file extensions.

**Comment by denisri** [comment]
> Finally, I'd like an expert to review a few points:
>
> the “capabilities” (in the DataSourceInfo sense)

They are here to tell if some partial or sequential reading can be done in an optimized way. As it is not supported here, this is OK.

> the storage_to_memory matrix (should it be 3-dimensional or more in case of higher-dimension images?)

IO generally fill a 3D matrix, but a larger one is now allowed. Indeed `Volume.flipToOrientation()` can flip to an orientation where other dimensions are moved, and then modifies the `storage_to_memory` matrix to a larger matrix if needed..

**Pull request merged**